vllm/model_executor/model_loader/loader.py

```diff
@@ -958,6 +989,12 @@ def _load_weights(self, model_config: ModelConfig,
                     f"BitsAndBytes loader does not support {quant_method} "
                     "quantization")
 
+        # The quant_states in pre_quantized models cannot work with a split
+        # weight tensor. So TP does not work with pre_quantized bnb models.
+        if pre_quant and get_tensor_model_parallel_world_size() > 1:
+            raise ValueError(
+                "Prequant BitsAndBytes models with TP is not supported.")
```

Comment:
This seems to be a big issue still as most people would be using prequantized models afaik

Reply:
In bnb, quantization turns a tensor into a tensor and a struct of quant_state. bnb does not allow de-quantization of a full-size quant_state working with a split quantized tensor. That makes me unable to implement TP directly on pre-quant models using the weights and quant_states out of box. 

I have been considering three solutions:
1. request the bnb community to support the above scenario. I am not sure how long it might take, given the speed of 
2. unpack the prequant models when loading and then treat it as non-quantized models. But the model loading will be terribly slow, which invalidates the purpose of pre-quantization.
3. suggest the user to turn to PP, as most popular models support PP now in vllm. I will update the error message.


@mgoin , what is your opinion?

Reply:
I see, this is unfortunate. I think you have chosen fairly in this case and the recommendation for PP in the error (#3) would be the best+simple

Reply:
Yep.